src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -465,10 +465,9 @@ type DiagnosticItem =
     static member Message(code: ErrorCode, args: IEnumerable<string>) =
         let message =
             match code with
-            | ErrorCode.TypeMismatch ->
-                "The type {1} does not match the type {0}.\nActual type:   {3}\nExpected type: {2}"
+            | ErrorCode.TypeMismatch -> "The type {0} does not match the type {1}.\nExpected: {2}\n  Actual: {3}"
```

Comment:
Does this make TypeMismatch a special case of the TypeIntersectionMismatch, where the first argument is 'match'?

Reply:
Sort of, but they don't quite mean the same thing by "expected type"... for TypeMismatch, the actual type is usually a subtype of the expected type. For TypeIntersectionMismatch, the actual type may be either a subtype or a supertype of the expected type. I changed the error messages a bit based on this.